cosmos/airflow/graph.py

```diff
@@ -78,15 +86,14 @@ def create_task_metadata(node: DbtNode, execution_mode: ExecutionMode, args: dic
             arguments=args,
         )
         return task_metadata
-    else:
-        logger.error(f"Unsupported resource type {node.resource_type} (node {node.unique_id}).")
+
+    raise ValueError(f"Unsupported resource type {node.resource_type} (node {node.unique_id}).")
```

Comment:
It could be more helpful to end-users if we were using a custom exception, as opposed to a Python one